script.js

```diff
@@ -1,12 +1,208 @@
+    // make card for each episode
+    makeCardForEpisodes(allEpisodes);
+  }
 }
 
-window.onload = setup;
+//Get all episodes via fetch
+async function getAllAvailableEpisodes(showId) {
+  // await response of the fetch call
+  const response = await fetch(
+    `https://api.tvmaze.com/shows/${showId}/episodes`
+  );
+  // only proceed once its resolved
+  const data = await response.json();
+  // only proceed once second promise is resolved
+  return data;
+}
+
+//make card for each episode
+function makeCardForEpisodes(episodeList) {
+  let output = "";
+  episodeList.forEach((episode) => {
+    output += `
+      <div class="col">
+        <div class="card h-100 px-0 mx-1">
+        <img src="${episode.image.medium}" class="card-img-top mx-0" alt="...">
+        <div class="card-body mb-4">
+        <h5 >${episode.name} - S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.season.toString().padStart(2, "0")}</h5>
+      <p>${episode.summary}</p>
+        </div>
+        <a href="${
+          episode.url
+        }" class="btn btn-primary" style="position:absolute; bottom:10px; left:20px; right:20px;" target="_blank">See More</a>
+      </div>
+      </div>
+    `;
+  });
+  showEpisodseRow.innerHTML = output;
+
+  // Create option for each episodes in select after loading page
+  episodeList.forEach((episode) => {
+    const option = document.createElement("option");
+    option.value = `S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")}`;
+    option.textContent = `S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")} - ${
+      episode.name
+    }`;
+    selectEpisodes.appendChild(option);
+  });
+}
+
+// select a specific episode
+function selectEpisode(event) {
+  if (event.target.value == "allEpisodes") {
+    searchUser.value = "";
+    searchUser.disabled = false;
+    //Remove options from episodes's select
+    const options = selectEpisodes.getElementsByTagName("option");
+    for (let i = options.length - 1; i > 0; i--) {
+      selectEpisodes.removeChild(options[i]);
+    }
+    makeCardForEpisodes(allEpisodes);
+  } else {
+    searchUser.value = "";
+    searchUser.disabled = true;
+    const season = event.target.value.slice(1, 3);
+    const number = event.target.value.slice(4, 6);
+    const selectedMovie = allEpisodes.filter(
+      (episode) => episode.season === season && episode.number === number
+    );
+    makeCardForEpisodes(selectedMovie);
+  }
+}
+
+// Show episodes based on live search
+function searchMovie(e) {
+  const selectedMovie = [];
+  // Get the value of input
+  let serchContent = e.target.value;
+  if (serchContent == "") {
+    makeCardForEpisodes(allEpisodes);
+    searchNumResult.textContent = "";
+  } else {
+    // Create case-insensitive RegExp
+    let searchContentInsensitive = new RegExp(serchContent, "i");
```

Comment:
Can you talk through why you decided to use regular expressions here? What other approaches could you have taken?

Reply:
can I use toLowerCase method ?

Reply:
Yeah, probably a `String.toLowerCase().contains` call - give it a go! :)